tests/lib/c_lib/src/main.c

```diff
@@ -624,10 +624,16 @@ ZTEST(test_c_lib, test_str_operate)
 
 	zassert_true(strncat(ncat, str1, 2), "strncat failed");
 	zassert_not_null(strncat(str1, str3, 2), "strncat failed");
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
+#endif
+#if defined(__GNUC__) && __GNUC__ >= 7
 #pragma GCC diagnostic ignored "-Wstringop-overflow"
+#endif
```

Comment:
```suggestion
#if defined(__GNUC__) && __GNUC__ >= 7
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wstringop-overflow"
#endif
```

These two should be kept together. After all, there is no point in doing push/pop unless you add ignore for something.

Reply:
Made this change too, and actually Googled "GCC diagnostic pragmas" this time! :sweat_smile: Thanks for your patience.